OrrRock/OrrRock/ViewControllers/HomeCardCollectionViewThumbnailCell.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  CardCollectionViewThumbnailCell.swift
+//  OrrRock
+//
+//  Created by Park Sungmin on 2022/10/19.
+//
+
+// MARK: TODO
+// HomeCollectionViewCardCell에서 setData() 함수를 통해 이미지를 받아와 thumbnailView에 이미지를 넣어주는 기능 구현 필요.
+// 혹은, URL 링크를 받아와, ThumbnailCell 내에서 해당 이미지를 호출하는 기능 구현 필요.
+
+import Foundation
+import SnapKit
+import UIKit
+
+class HomeCardCollectionViewThumbnailCell: UICollectionViewCell {
+    let thumbnailView: UIImageView = {
+        let view = UIImageView()
+        return view
+    }()
+    
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        setLayout()
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: .zero)
+        setLayout()
+    }
+    
+    
+    func setLayout() {
+        contentView.addSubview(thumbnailView)
+        thumbnailView.backgroundColor = .systemGray
+        
+        thumbnailView.snp.makeConstraints { make in
+            make.edges.equalTo(contentView.safeAreaLayoutGuide)
+        }
```

Comment:
```suggestion
        thumbnailView.snp.makeConstraints { 
            $0.edges.equalTo(contentView.safeAreaLayoutGuide)
        }
```

이렇게 해도 되는 스냅킷의 마법! 🔥

Reply:
```swift
// 함수의 정의
func performClosure(param: (String) -> Int) {
    param("Swift")
}

// 1) 타입 추론(Type Inference)
performClosure(param: { (str: String) in
    return str.count
})

performClosure(param: { str in
    return str.count
})

// 2) 싱글 익스프레션인 경우(코드가 한줄인 경우), 리턴을 안 적어도 됨(Implicit Return)
performClosure(param: { str in
    str.count
})

// 3) 아규먼트 이름을 축약(Shorthand Argements)
performClosure(param: {
    $0.count  // 두번째 파라미터: $1 , 세번째 파라미터: $2
})

// 4) 트레일링 클로저
performClosure(param: {
    $0.count
})

performClosure() {
    $0.count
}

performClosure { $0.count }  // 한줄로 표현이 가능해짐
```

사실 스냅킷이 아니더라도 스위프트 클로저의 마법!!